### DISCUSSION TITLE
Location data is not being sent properly

### DISCUSSION THREAD

**Discussion by Leroing** [aerostack2/aerostack2#657] [Q&A]
The current issue with the project is that the location data is not being sent to /cf1/self_localization/pose and /cf2/self_localization/pose. On the other hand we find that everything works properly for cf0. When running mocap.launch.py we only get an output on /cf0/self_localization/pose when the namespace is cf0. When the namespace is set to anything like cf1 we receieve no output on any of the self localization poses. The rigidbodies and ground truth topics work fine. We can ros2 topic echo to see that all the data is being passed properly. The issue lies in the last level of the system only redirecting output to /cf0/self_localization/pose and not to cf1 or cf2.

**Comment by miferco97** [comment]
Can you provide the config files that you are using for launching the project?

**Reply by zZ-kunja-Zz** [reply]
Heres a list of the relevant configuration files from the project, let me know if you need more files from our project

swarm_config_file .yaml
motion_controller.yaml
platform_config_mocap.yaml
cf0_state_estimator.yaml
cf1_state_estimator.yaml
optitrack.yaml
swarm_config_file.yaml
platform_config_file.yaml
mocap.yaml

1. swarm_config_file.yaml
yaml
Copy code
cf0:
  uri: radio://0/80/1M/E7E7E7E701
cf1:
  uri: radio://0/80/1M/E7E7E7E702
cf2:
  uri: radio://0/80/1M/E7E7E7E703


2. motion_controller.yaml
yaml
Copy code
/**:
  ros__parameters:
    cmd_freq: 10.0  # Hz of platform commands send standard is 100
    info_freq: 10.0  # Hz of platform info publish standard is 100
    odom_frame_id: "odom"
    base_frame_id: "base_link"
    use_bypass: true


3. platform_config_mocap.yaml
yaml
Copy code
/**:
  ros__parameters:
    cmd_freq: 10.0  # Hz of platform commands send standard is 100
    info_freq: 10.0  # Hz of platform info publish standard is 100
    external_odom: true  # Availability of external odometry
    external_odom_topic: "self_localization/pose" # External odometry topic name self_localization/pose
    controller_type: 1  # Controller type Any(0), PID(1), Mellinger(2), INDI(3)
    estimator_type: 2 # Estimator type Any(0), complementary(1), kalman(2)


4. cf0_state_estimator.yaml
yaml
Copy code
/**:
  ros__parameters:
    use_ignition_tf: True
    rigid_body_name: "1"
    mocap_topic: "mocap/rigid_bodies"  #/ground_truth/pose"
    twist_smooth_filter_cte: 0.1
    orientation_smooth_filter_cte: 0.1


5. cf1_state_estimator.yaml
yaml
Copy code
/**:
  ros__parameters:
    use_ignition_tf: True
    rigid_body_name: "2"
    mocap_topic: "mocap/rigid_bodies"
    twist_smooth_filter_cte: 0.1
    orientation_smooth_filter_cte: 0.1


6. optitrack.yaml
yaml
Copy code
/**:
  ros__parameters:
    connection_type: "Multicast" # Unicast / Multicast
    server_address: "192.168.68.50"
    local_address: "192.168.68.52"
    multicast_address: "239.255.42.99"
    server_command_port: 1510
    server_data_port: 1511


7. swarm_config_file.yaml
yaml
Copy code
/cf1:
  platform:
    ros__parameters:
      uri: "radio://0/1/1M/E7E7E7E701"
  aideck_pub:
    ros__parameters:
      cam:
        ip: "192.168.0.109"
        port: 5000   
        calibration_file: "camera_calibration.yaml"

/cf2:
  platform:
    ros__parameters:
      uri: "radio://0/1/1M/E7E7E7E702"

/cf3:
  platform:
    ros__parameters:
      uri: "radio://0/1/1M/E7E7E7E703"


8. platform_config_file.yaml
yaml
Copy code
/**:
  ros__parameters:
    cmd_freq: 100.0  # Hz of platform commands send
    info_freq: 100.0  # Hz of platform info publish
    external_odom: true  # Availability of external odometry
    external_odom_topic: "self_localization/pose" # External odometry topic name
    controller_type: 1  # Controller type Any(0), PID(1), Mellinger(2), INDI(3)
    estimator_type: 2 # Estimator type Any(0), complementary(1), kalman(2)
    multi_ranger_deck: false  # Availability of multi ranger deck


9. mocap.yaml
yaml
Copy code
#
# Definition of all trackable objects
# Identifier corresponds to Trackable ID set in Tracking Tools

/**:
    ros__parameters:
        rigid_bodies:
            1:
                pose: "ground_truth/pose"
                pose2d: "Robot_1/ground_pose"
                child_frame_id: ""
                parent_frame_id: ""
            2:
                pose: "ground_truth/pose2"
                pose2d: "Robot_2/ground_pose"
                child_frame_id: ""
                parent_frame_id: ""
            3:
                pose: "ground_truth/pose3"
                pose2d: "Robot_3/ground_pose"
                child_frame_id: ""
                parent_frame_id: ""
        optitrack_config:
                multicast_address: "239.255.42.99"
                command_port: 1510
                data_port: 1511`
